experimental/oak_baremetal_channel/src/message.rs

```diff
@@ -0,0 +1,151 @@
+//
+// Copyright 2022 The Project Oak Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+use alloc::vec::Vec;
+
+pub type Length = u32;
+pub const LENGTH_SIZE: usize = 4;
+pub const LENGTH_OFFSET: usize = 0;
+static_assertions::assert_eq_size!([u8; LENGTH_SIZE], Length);
+
+pub type InvocationId = u32;
+pub const INVOCATION_ID_SIZE: usize = 4;
+pub const INVOCATION_ID_OFFSET: usize = 4;
+static_assertions::assert_eq_size!([u8; INVOCATION_ID_SIZE], InvocationId);
+
+pub type MethodId = u32;
+pub const METHOD_ID_SIZE: usize = 4;
+pub const METHOD_ID_OFFSET: usize = 8;
+static_assertions::assert_eq_size!([u8; METHOD_ID_SIZE], MethodId);
+
+pub type StatusCode = u32;
+pub const STATUS_CODE_SIZE: usize = 4;
+pub const STATUS_CODE_OFFSET: usize = 8;
+static_assertions::assert_eq_size!([u8; STATUS_CODE_SIZE], StatusCode);
+
+pub const PADDING_SIZE: usize = 4;
+
+pub const BODY_OFFSET: usize = 16;
+
+pub trait Message {
+    fn len(&self) -> usize;
+    fn encode(self) -> Vec<u8>;
+    fn decode(frames: Vec<u8>) -> Self;
+}
+
+#[derive(Debug, PartialEq, Eq, Clone)]
+pub struct RequestMessage {
+    pub invocation_id: InvocationId,
+    pub method_id: MethodId,
+    pub body: Vec<u8>,
+}
```

Comment:
Introduce separate structs for request- and response messages, matching the specification. There's plenty of overlap in their encoding implementation, but imo that's ok (both are tested) and means we no longer have to deal with the somewhat ugly `method_id_or_status` of past days.